### DISCUSSION TITLE
WebKit support

### DISCUSSION THREAD

**Discussion by Nwgazzazamwal** [MasterKale/SimpleWebAuthn#285] [General]
hello, any idea if this is supported inside a webview react native

**Comment by MasterKale** [comment]
I haven't done any native iOS dev myself, but my current understanding is that you want to use [SFSafariViewController](https://developer.apple.com/documentation/safariservices/sfsafariviewcontroller) or [ASWebAuthenticationSession](https://developer.apple.com/documentation/authenticationservices/aswebauthenticationsession) if you want to use passkeys in your native app.

The best advice I have for figuring out how to incorporate these into a React Native project, unfortunately, is to google "react native SFSafariViewController" or "react native ASWebAuthenticationSession". Good luck!

I think at a high level there's nothing preventing you from using **@simplewebauthn/browser** in your React Native project. Your biggest hurdle is to figure out how to get a webview that'll support WebAuthn so that the browser library can do its thing. That's what **SFSafariViewController** should get you.

If you're trying to handle WebAuthn-based auth entirely in native code, though, and are aiming to get a server set up to handle responses then you can use **@simplewebauthn/server** to parse **ASWebAuthenticationSession** responses no problem.

**Comment by MasterKale** [comment]
I'm going to turn this into a Discussion for now as it's not reporting an issue with SimpleWebAuthn.

**Comment by mahnunchik** [comment]
It strange but Android WebView has no support for WebAuthn at all (in almost 2025 year). https://github.com/apache/cordova-android/issues/1743

It seems that this affects all cordova and react native projects that use WebView.

iOS embedded WebView fully supports the WebView protocol, only the associated domains must be added correctly https://developer.apple.com/documentation/xcode/supporting-associated-domains

**Reply by Climax777** [reply]
https://developer.android.com/identity/sign-in/credential-manager-webview

They provide a workaround. We are going to test this with SimpleWebAuthn soon, but it seems that they provide a monkey-patch for the low-level scripts, which should then just work as a drop-in replacement.
